File: archive/e/elixir/even_odd.ex
defmodule EvenOdd do
    require Integer

    @doc """
    Determine whether a supplied number is Even or Odd.
    Return an error-string if the argument is missing or invalid. 
    """
    @spec main(argv :: list( String.t())) :: String.t()
    def main(argv) do
        if length(argv) != 1 do
            "Usage: please input a number"
        else 
            case argv |> List.first |> Integer.parse do
                {number, ""} -> even_odd_string(number)
                {_, _rest} -> "Usage: please input a number"
                :error -> "Usage: please input a number"
            end
        end
    end

    @spec even_odd_string(number :: integer) :: String.t()
    defp even_odd_string(number) do
        case Integer.is_even(number) do
            true -> "Even"
            false -> "Odd"
        end
    end
end

EvenOdd.main(System.argv()) |> IO.puts

File: archive/t/typescript/even-odd.ts
const error_msg: string = "Usage: please input a number"
let num_str:string = (process.argv.length == 3) ? process.argv[2] : "";
let num:number = parseInt(num_str);
if (isNaN(num)) {
    console.log(error_msg);
    process.exit(1);
}
let is_even:boolean = (num % 2 == 0);
console.log(is_even ? "Even" : "Odd");